pyproject.toml

```diff
@@ -34,21 +33,47 @@ colorama = "^0"
 jsonschema = "^4"
 types-colorama = "^0"
 types-requests = "^2"
+bs4 = "^0"
+
+transformers = { version = "^4", optional = true }
```

Comment:
I wonder, how will transformers etc **not** go into the main dependency list if it is listed here along with bumpy etc?

Reply:
From https://packaging.python.org/en/latest/tutorials/installing-packages/

> Extras are optional “variants” of a package, which may include additional dependencies, and thereby enable additional functionality from the package.

which means if we do not add `huggingface-agent` in `pip install .[huggingface-agent]` or `poetry install -E huggingface-agent`, they will not be installed into the environment. You can try it out to confirm.

Reply:
Brilliant! Thanks for digging up the way to make it happen! Please add what you've just exploit to me about `pip install .[huggingface-agent]` to the instruction. I believe this syntax is more familiar to developers than `-E key`.

Reply:
Ah, sorry, the instruction is there. Awesome!